packages/core/upload/server/services/metrics/index.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const getProviderName = () => strapi.config.get('plugin.upload.provider', 'local');
+const getProviderIsPrivate = () => strapi.plugin('upload').provider.isPrivate();
```

Comment:
isPrivate might not be define in a provider 🤔. If it has to be async make sure this function is async too. or remove the await below :)

Reply:
we are defined in the provider in the private-s3-bucket feature branch, by default it returns null.

I will change this function to be async !